synapse/handlers/federation_event.py

```diff
@@ -1315,6 +1318,47 @@ async def _handle_marker_event(self, origin: str, marker_event: EventBase) -> No
             marker_event,
         )
 
+    async def backfill_event(
+        self, destination: str, room_id: str, event_id: str
+    ) -> None:
+        logger.info("backfill_event event_id=%s", event_id)
+
+        room_version = await self._store.get_room_version(room_id)
+
+        event_from_response = await self._federation_client.get_pdu(
+            [destination],
+            event_id,
+            room_version,
+        )
+
+        if not event_from_response:
+            raise FederationError(
+                "ERROR",
+                404,
+                "Unable to find event_id=%s from destination=%s to backfill."
+                % (event_id, destination),
+                affected=event_id,
+            )
+
+        # We want to make a non-outlier event so it plays well with
+        # `_process_pulled_events()` -> `_update_outliers_txn()` to create a
+        # `state_group` and mimics what would happen in a regular backfill.
+        # `get_pdu()` can potentially return an `outlier` depending on the cache
+        # which we don't want.
+        event_non_outlier = make_event_from_dict(
+            event_from_response.get_pdu_json(),
+            event_from_response.room_version,
+            internal_metadata_dict=None,
+        )
+        assert not event_non_outlier.internal_metadata.outlier
```

Comment:
See https://github.com/matrix-org/synapse/pull/13205#discussion_r918365746 for an alternative to handle this in the `_process_pulled_events` stack instead

Reply:
> maybe it should happen in get_pdu instead?

Good suggestion 👍

I've updated `get_pdu` to always return a copy of the event so that it's cache can stay pristine with the original PDU.

Reply:
`get_pdu` changes split out to https://github.com/matrix-org/synapse/pull/13320